catalyst_voices/packages/internal/catalyst_voices_models/lib/src/document/document_definitions.dart

```diff
@@ -116,6 +109,10 @@ sealed class BaseDocumentDefinition extends Equatable {
     final ref = refPath.split('/').last;
     return refPathToDefinitionType[ref] != null;
   }
+
+  T? castValue(Object? value);
+
+  DocumentProperty<T> castProperty(DocumentProperty<T> property);
```

Comment:
We are casting from an object to type T, probably my mistake when I provided this sample, it should be:

```suggestion
  T? castValue(Object? value);

  DocumentProperty<T> castProperty(DocumentProperty<Object> property);
```